### DISCUSSION TITLE
HTTP prior acknowledgement

### DISCUSSION THREAD

**Discussion by claypot-ai** [mojolicious/mojo#1844] [Q&A] [answered]
I have this perl code:

post '/api/prep/' => sub {
.
.
$self->render(json => {"status"=>"200 OK", "message"=>"acknowledged", "state"=>"PREP"});
.
.<<some codes to perform an operation work that will take at least 10mins to complete>>
.
close $ofh ;
}

but the json is returned to the requester who did the HTTP POST after the entire operation work is completed. 

As the operation work takes at least 10mins to complete, how can i return to the requester the json before the operation work begins so that the requestor knows that there is a acknowledgement response and not wait 10mins later for the operation work to complete?

**Comment by fskale** [comment, marked as answer]
10 Minutes ?
You really should use a job queue for this particular purpose.
https://docs.mojolicious.org/Minion/

**Reply by kraih** [reply]
Agreed.

**Reply by claypot-ai** [reply]
What if i don't use Minon and instead code it to return a HTTP response prior to the completion of the operation?

How do i get the requestor's IP and TCP port so that i can send a HTTP response?

**Reply by Grinnz** [reply]
You can't. You can only send one response to a request, and the client must send an additional request for another response. Alternatively, the client can connect to a websocket to hold open a persistent connection.

**Reply by kraih** [reply]
And you have to consider non-blocking I/O for everything you do.
